osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -39,26 +43,47 @@ public class SkinProvidingContainer : Container, ISkinSource
 
         protected virtual bool AllowColourLookup => true;
 
+        /// <summary>
+        /// Constructs a new <see cref="SkinProvidingContainer"/> with a single skin added to the protected <see cref="SkinSources"/> list.
+        /// </summary>
         public SkinProvidingContainer(ISkin skin)
+            : this()
         {
-            this.skin = skin;
+            SkinSources.Add(skin);
+        }
 
+        /// <summary>
+        /// Constructs a new <see cref="SkinProvidingContainer"/> with no sources.
+        /// Up to the implementation for adding to the <see cref="SkinSources"/> list.
+        /// </summary>
+        protected SkinProvidingContainer()
+        {
             RelativeSizeAxes = Axes.Both;
 
             noFallbackLookupProxy = new NoFallbackProxy(this);
 
-            if (skin is ISkinSource source)
-                source.SourceChanged += TriggerSourceChanged;
+            SkinSources.BindCollectionChanged(((_, args) =>
+            {
+                switch (args.Action)
```

Comment:
Since this is doing quite dangerous stuff (binding to external events), I'd like to see the rest of the switch cases handled, such as `Reset` and `Replace`.

Reply:
I was always wondering why some usages don't handle all cases other than `Add` and `Remove`, but TIL.